### ISSUE TITLE
Git branch shows master with --git-branch set 

### ISSUE THREAD

**Issue by wmiller112** [fluxcd/flux#2978]
**Describe the bug**

Git branch in flux pod always `master`, despite --git-branch being set. It applies files from the appropriate branch, but always shows master. This results in issues with .gitattribute filters, which I am using to set namespace to the branch name in a manifest file.

**To Reproduce**

Steps to reproduce the behaviour:

1. Install Flux via helm chart (tested 1.2.0 and 1.3.0) with git.branch set to any branch other than master
2. Flux clones repo, and attempts to apply manifests from the correct branch
3. Exec'ing in to flux pod, /tmp/flux-gitclone####,`git rev-parse --abbrev-ref HEAD` shows `master`, and the .gitattributes filter (set via config.data) that uses branch name is subs `master`

**Expected behavior**

`git rev-parse --abbrev-ref HEAD` shows the branch set as --git-branch, and subsequently, .gitattributes filter to sub the correct branch name rather than master.

**Logs**

If applicable, please provide logs of `fluxd`. In a standard stand-alone installation of Flux, you'd get this by running `kubectl logs deploy/flux  -n flux`.

**Additional context**

- Flux version: 1.19.0
- Kubernetes version: 1.15

**Comment by asubmani** [comment]
+1 
This issue was driving me crazy for over 3 hrs. @wmiller112  Do you have a workaround?  I am exploring using fluxctl or kubectl with yamls... but all examples seem to cover "Personal Authentication Tokens" with these tools OR have flux create the ssh keys during initialization.

**Comment by wmiller112** [comment]
So just for context, the issue this was causing for me was that I was trying to use a gitattributes smudge/clean filter to set the image filter annotation in the files I use to generate helmrelease. This is in an effort to support a dynamic, on-demand staging configuration by reusing the same environment (staging) helmrelease template files. Ideally, I'd like to just grab the branch name via git, but since I've automated the process to create my flux deployments (via the helm chart, deployed with terraform), and I needed a script to generate some config files anyway, I ended up just having the script generate the filter command by passing in the branch name which the requester is providing manually as the desired branch to deploy anyway. Definitely not the most elegant solution, but got me unblocked.
